chiselFrontend/src/main/scala/chisel3/core/Data.scala

```diff
@@ -208,6 +208,13 @@ abstract class Data extends HasId {
     }
   }
 
+  // If this Data is an instance of an inner class, record enclosing class
+  // This is only used for cloneType!
+  protected var outerModule: Option[BaseModule] =
+    (Option(this.getClass.getEnclosingClass) zip Builder.currentModule)
```

Comment:
Is this just the odd functional one-liner version of saying if this has an enclosing class and the current Builder module is assignable to it, then set that to the outer module?

Reply:
Yes, just using option to elide multiple null checks